test/smoke/testApps/AutoPerfCounters/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PerfCountersDataTest.java

```diff
@@ -0,0 +1,163 @@
+        long start = System.currentTimeMillis();
+        List<Envelope> performanceItems =
+                // this is the new method
+                mockedIngestion.waitForItems(new Predicate<Envelope>() { // this Predicate object lets you check for the
+                    // item you are looking for.
+                    @Override
+                    public boolean apply(@Nullable Envelope input) {
+                        if (input == null) {
+                            return false;
+                        }
+                        // here, we are just checking if the data type == "PerformanceCounterData", but you could check for more conditions.
+                        boolean rval = "PerformanceCounterData".equals(input.getData().getBaseType());
+                        return rval;
+                    }
+                }, 8, // waits for 8 items
+                        150, TimeUnit.SECONDS); // times out after 150 seconds (it will throw a TimeoutException)
+
+        assertEquals(8, performanceItems.size());
+        for (Envelope item : performanceItems) {
+            assertEquals("PerformanceCounterData", item.getData().getBaseType());
+        }
+        System.out.println("PerformanceCounterData are good: " + (System.currentTimeMillis() - start));
+
+        System.out.println("Waiting for metric data...");
+        List<Envelope> metricItems = mockedIngestion.waitForItems(new Predicate<Envelope>() {
+            @Override
+            public boolean apply(@Nullable Envelope input) {
+                if (input == null) {
+                    return false;
+                }
+                boolean rval = "MetricData".equals(input.getData().getBaseType());
+                return rval;
+            }
+        }, 4, 10, TimeUnit.SECONDS);
+        assertEquals(4, metricItems.size());
+        for (Envelope item : metricItems) {
+            assertEquals("MetricData", item.getData().getBaseType());
+        }
+        System.out.println("MetricData are good: " + (System.currentTimeMillis() - start));
+
+        System.out.println("Waiting for requests...");
+        start = System.currentTimeMillis();
+        Envelope requestItem = mockedIngestion.waitForItem(new Predicate<Envelope>() {
+            @Override
+            public boolean apply(@Nullable Envelope input) {
+                if (input == null)
+                    return false;
+                return "RequestData".equals(input.getData().getBaseType());
+            }
+        }, 10, TimeUnit.SECONDS); // this one just returns a single envelope.
+        // at this point, the item has already arrived and this just returns immediately
+        assertNotNull(requestItem);
+        System.out.println("Requests are good: " + (System.currentTimeMillis() - start));
+
+        int heapMemoryUsed = 0;
+        int suspected = 0;
+        for (int i = metricItems.size(); i > 2; i = i - 1) {
+            DataPoint dp1 = getMetricDataDetails(i - 1);
```

Comment:
@Terryxxx, we need to verify that each of the `MetricData` items have the property `"CustomPerfCoutner"` set to `"true"`.
e.g. `myEnvelope.getData().getProperties().get("CustomPerfCounter")` should return `"true"` for some `Envelope myEnvelope`.

Reply:
@littleaj ,I have changed the code according to your comments. Please review again. Thanks.